src/Microsoft.DotNet.Build.Tasks.Feed/BlobFeedAction.cs

```diff
@@ -89,6 +98,22 @@ public BlobFeedAction(string expectedFeedUrl, string accountKey, MSBuild.TaskLog
                     catch (InvalidOperationException ex) when (ex.Message.Contains("init"))
                     {
                         Log.LogWarning($"Sub-feed {source.FeedSubPath} has not been initialized. Initializing now...");
+                        requiresInit = true;
+                    }
+                    catch (InvalidOperationException ex) when (ex.Message.Contains("Unable to obtain a lock on the feed."))
+                    {
+                        Log.LogWarning($"Sleet was not able to get a lock on the feed. Sleeping {delay} seconds and retrying.");
+                        await Task.Delay(delay);
+
+                        // Pushing packages might take more than just 60 seconds, so on each iteration we add the defined value to itself so wait for
+                        // a bit more the next iterations
+                        delay += delay;
+                    }
+
+                    // If the feed has not been Init'ed this will be caught in the first iteration
+                    if (requiresInit && i == 0)
+                    {
+                        i--;
                         bool result = await InitAsync();
```

Comment:
(Duplicating outdated comment to make it visible again:)

> Does Init ever fail with an exception that should cause a retry?

Reply:
We want to fail fast if this fails. I've added return code to it